google/cloud/forseti/services/inventory/storage.py

```diff
@@ -930,6 +931,32 @@ def get_latest_inventory_index_id(cls, session):
             inventory_index.id)
         return inventory_index.id
 
+    @classmethod
+    def get_inventory_index_id_by_scanner_index_id(cls,
+                                                   session,
+                                                   scanner_index_id):
+        """List all inventory index entries.
+
+        Args:
+            session (object): Database session
+            scanner_index_id (int): id of the scanner in scanner_index table
+
+        Returns:
+            int64: inventory index id
+        """
+
+        inventory_index = (
+            session.query(InventoryIndex).filter(
```

Comment:
A couple of comments, where this query should be much simpler:
1: We should query on the `ScannerIndex` table.
2: We should not care about the `inventory_status`, since there is a 1:1 relationship of the `scanner_index_id` to `inventory_index_id`, which is the contract of what this method is suppose to return.

What do you think?

Reply:
agree!